packages/types/src/amount.ts

```diff
@@ -26,7 +26,9 @@ export const fromValueView = (valueView: ValueView): BigNumber => {
 };
 
 export const fromString = (amount: string, exponent = 0): Amount =>
-  new Amount(toBaseUnit(BigNumber(amount), exponent));
+  new Amount(
+    toBaseUnit(BigNumber(amount).decimalPlaces(exponent, BigNumber.ROUND_FLOOR), exponent),
```

Comment:
Without this rounding, JS throws because we're trying to convert a decimal to a `BigInt`.